### ISSUE TITLE
Starting masternodes from the masternode tab fails with "Error: invalid IP"

### ISSUE THREAD

**Issue by Zkffkah** [wagerr/wagerr#6]
same problem as https://github.com/PIVX-Project/PIVX/issues/373, and is solved upstream

**Comment by Cryptarchist** [comment]
@lowwor thank you for looking into this, I have opened #7 for review.

**Comment by WagerrTor** [comment]
thanks lowwor for creating the issue.

I have no issues and can reproduce it only by wrong/not specified settings which does not make it to a bug. Everything works correctly on my side. I see that in other project they have implemented it, but I miss also there some input for this issue. If your external IP is correct, then you will have no issues. If you start cold with 127.0.0.0/24 there are also no issues.

Try specifying it in your config, something like this:
```
...
listen=0
externalip=YourExternalIP
bind=YourExternalIP
...
```

If you run masternodes in multi mode you do not need to specify masternode address and priv key
```
masternode=0
```

, in single mode you need to do it:
```
masternode=1
masternodeaddr=YourExternalIp:55002
masternodeprivkey=7hLgSjJnKdnsDMPz3ed8cKuWjrQi9DuMWYHXBi9DtXRyNE4hBF4 # key which is created with masternode genkey. Note: it is non compressed key, you can recognise it on private key starting with 7
```

Can you retry with settings I posted?

**Comment by Zkffkah** [comment]
Yeah, I have those settings in my server's wagerr.conf, something like this.
```
rpcuser=gaegeargear
rpcpassword=aefaefaerf
rpcallowip=127.0.0.1
listen=0
server=1
daemon=1
logtimestamps=1
maxconnections=256
masternode=1
externalip=37.98.67.45
bind=37.98.67.45
masternodeaddr=37.98.67.45:55002
masternodeprivkey=awefwaefafwaefwaefawefwa
```
Sorry I don't describe the detail. The problem occur only when I click the start alias in my local wallet with the gui start alias button. It's ok with the debug console command.
![cache_5ce38d292e939200](https://user-images.githubusercontent.com/5444033/36630566-91c443ba-19a3-11e8-8448-079e09e044e3.jpg)

**Comment by Kokary** [comment]
Discussion on this from Bitcoin: https://github.com/bitcoin/bitcoin/pull/7553 and here: https://github.com/bitcoin/bitcoin/issues/7098

How does this solve the issue?

The PR seems like a good idea anyway.

**Comment by WagerrTor** [comment]
@lowwor 
Can you tell me also which client do you use and if this issue is resolved for you in the meantime?

**Comment by Zkffkah** [comment]
I am using v1.4.34.2 which the problem still exists. I will check with the new version when it's available with #7 in it.

**Comment by Zkffkah** [comment]
Sorry for the delay confirmation, problem solved. Nice.
